examples/client_test.cpp

```diff
@@ -82,6 +82,11 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "session_view.hpp"
 #include "print.hpp"
 
+#if TORRENT_USE_I2P
+// this is cheating, for base32encode_i2p()
```

Comment:
Does not work for me.
_(However, I'm not sure if it's not because of my mistake)_

ld.lld: error: undefined symbol: libtorrent::base32encode_i2p(libtorrent::span<char const>)
&gt;>> referenced by examples/CMakeFiles/client_test.dir/client_test.cpp.obj...

Seems like adding
```
	../src/escape_string.cpp
	../src/hex.cpp
	../src/parse_url.cpp
	../src/string_util.cpp
	../src/random.cpp
```
to CMakeLists.txt helped.

Reply:
client_test.cpp is currently "cheating", it's using a non-public function (`base32encode_i2p`) from libtorrent. If you build libtorrent as a DLL on windows, that function won't be exported and you'll probably see a link error. It should probably be fixed before landing. maybe client_test doesn't need to print addresses as base32